src/util/mmap_cache.h

```diff
@@ -146,6 +146,14 @@ struct sss_mc_initgr_data {
                              * after gids */
 };
 
+struct sss_mc_sid_data {
+    rel_ptr_t name;         /* ptr to SID string, rel. to struct base addr */
+    uint32_t type;          /* enum sss_id_type */
+    uint32_t id;            /* gid or uid */
+    uint32_t sid_len;       /* length of sid */
+    char sid[0];
```

Comment:
I need to educate myself here. Why we use `char sid[0]` instead of `char *sid`?

Reply:
To be able to conveniently keep everything in a single continuous chunk of memory.

Memory layout:  chunk = { 4 bytes (field 1), 4 bytes (field 2), ..., 4 bytes (field N), M bytes (string) }

Then:
data = (sss_mc_sid_data *) chunk;
Addr of `data->sid` is the beginning of a string.

But one can't have a real pointer with a "real" address here - when client process mmap()-s this memory chunk into own virtual address space, value of this pointer (populated in sssd_nss address space) wouldn't have any meaning.
You can have only "relative" pointer - and this is exactly field `name` in this structure.
Practically (data + data->name) and (&data->sid) points to the same memory.